### ISSUE TITLE
--rack argument isn't always respected

### ISSUE THREAD

**Issue by dlanger** [convox/rack#1107]
There are a number of `convox` commands which are documented (according to `convox help`) to accept the `--rack` argument, but which silently ignore it and use the current rack.

Here's an example with the `convox ssl` command. In it, I:
1. Log into the `stage` rack using `convox switch`
2. Update the SSL certificate on the `stage` rack
3. Try and update the SSL certificate on the `kraken` rack, using the `--rack` argument - but get an error saying there's nothing to be done
4. Switch to the `kraken` rack using `convox switch`
5. Issue the same command as in #3, but in this case the update is successful

I'd expect #2 to work, as I'm using the `--rack` command to specify which rack I'd like the command to impact.

```
$ convox --version
client: 20160818150650
$ convox switch
wave/stage
$ convox ssl update main:443 acm-9b81e07ebc40
Updating certificate... OK
$ convox ssl --rack kraken update main:443 acm-9b81e07ebc40
Updating certificate... ERROR: ValidationError: No updates are to be performed.
        status code: 400, request id: b445b025-6663-11e6-bbe2-a10804931323
$ convox switch kraken
Switched to wave/kraken
$ convox ssl --rack kraken update main:443 acm-9b81e07ebc40
Updating certificate... OK
```

I've seen this in a few other places which I don't recall - I'll update if I find them again.

**Comment by dlanger** [comment]
Also #1202

**Comment by dlanger** [comment]
The order in which you use the `--rack` is also meaningful, and doesn't provide helpful feedback when you do it wrong.

As an example, this:
```
convox api --rack RACKNAME get /apps/APPNAME
```
and
```
convox api get /apps/APPNAME --rack RACKNAME
```
do different things.

The first one makes the API call against whatever your current rack is, and ignores the `--rack` argument passed in (and doesn't tell you that it's being ignored). The second one makes the API call against `RACKNAME`.

**Comment by soulshake** [comment]
This should be resolved by #1684, please let us know if you encounter the misbehavior again.
